src/litdata/streaming/reader.py

```diff
@@ -65,8 +62,9 @@ def __init__(
         self._chunks_index_to_be_deleted: List[int] = []
         self._max_cache_size = max_cache_size
         self._parent_cache_dir = os.path.dirname(self._config._cache_dir)
-        self._to_download_queue: multiprocessing.Queue = multiprocessing.Queue()
-        self._to_delete_queue: multiprocessing.Queue = multiprocessing.Queue()
+        self._to_download_queue: Queue = Queue()
+        self._to_delete_queue: Queue = Queue()
+        self._stop_event = Event()
```

Comment:
The `Event` is better than the `_EOF_TOKEN`. Because when `if self._pre_download_counter < self._max_pre_download` is `True`, the `_EOF_TOKEN` will never be read.